src/coordinator/ts/m3db/block/convertor_test.go

```diff
@@ -0,0 +1,178 @@
+			{},
+			{Timestamp: now.Add(17 * time.Minute), Value: 5},
+			{Timestamp: now.Add(19 * time.Minute), Value: 6},
+		}),
+	}
+
+	seriesTwoBlockOne := newSeriesBlock(seriesIteratorsTwo[0], now, now.Add(10*time.Minute))
+	seriesTwoBlockTwo := newSeriesBlock(seriesIteratorsTwo[1], now.Add(10*time.Minute), now.Add(20*time.Minute))
+	tagsTwo, err := ident.NewTagStringsIterator(rawTagsTwo...)
+	require.NoError(t, err)
+
+	seriesTwo := SeriesBlocks{
+		Blocks: []SeriesBlock{seriesTwoBlockOne, seriesTwoBlockTwo},
+		ID:     ident.StringID("test_two"),
+		Tags:   tagsTwo,
+	}
+
+	return []SeriesBlocks{seriesOne}, []SeriesBlocks{seriesTwo}
+}
+
+func newMockIterator(t *testing.T, id string, tags []string, ctrl *gomock.Controller, dps []ts.Datapoint) encoding.SeriesIterator {
+	seriesIterator := encoding.NewMockSeriesIterator(ctrl)
+
+	seriesIterator.EXPECT().ID().Return(ident.StringID(id))
+
+	tagIter, err := ident.NewTagStringsIterator(tags...)
+	require.NoError(t, err)
+	seriesIterator.EXPECT().Tags().Return(tagIter)
+
+	seriesIterator.EXPECT().Next().Return(true).Times(len(dps))
+	seriesIterator.EXPECT().Next().Return(false)
+
+	for _, dp := range dps {
+		seriesIterator.EXPECT().Current().Return(ts.Datapoint{Timestamp: dp.Timestamp, Value: dp.Value}, xtime.Millisecond, nil).Times(1)
+	}
+
+	return seriesIterator
+}
+
+func TestConvertM3Blocks(t *testing.T) {
+	now := time.Now()
+	ctrl := gomock.NewController(t)
+	seriesOne, seriesTwo := newMultiNamespaceSeries(ctrl, now, t)
+	multiNamespaceSeriesList := []MultiNamespaceSeries{seriesOne, seriesTwo}
+	m3CoordBlocks, err := SeriesBlockToMultiSeriesBlocks(multiNamespaceSeriesList, nil, time.Minute)
+	require.NoError(t, err)
+
+	assert.Equal(t, "test_one", m3CoordBlocks[0].Blocks[0].ConsolidatedNSBlocks[0].SeriesIterators.Iters()[0].ID().String())
+	assert.Equal(t, models.Tags{"foo": "bar", "same": "tag"}, m3CoordBlocks[0].Blocks[0].Metadata.Tags)
+	assert.Equal(t, "test_two", m3CoordBlocks[0].Blocks[1].ConsolidatedNSBlocks[0].SeriesIterators.Iters()[0].ID().String())
+	assert.Equal(t, models.Tags{"biz": "baz", "same": "tag"}, m3CoordBlocks[0].Blocks[1].Metadata.Tags)
+	assert.Equal(t, now, m3CoordBlocks[0].Metadata.Bounds.Start)
+	assert.Equal(t, now.Add(10*time.Minute), m3CoordBlocks[0].Metadata.Bounds.End)
+
+	assert.Equal(t, "test_one", m3CoordBlocks[1].Blocks[0].ConsolidatedNSBlocks[0].SeriesIterators.Iters()[0].ID().String())
+	assert.Equal(t, "test_two", m3CoordBlocks[1].Blocks[1].ConsolidatedNSBlocks[0].SeriesIterators.Iters()[0].ID().String())
+	assert.Equal(t, now.Add(10*time.Minute), m3CoordBlocks[1].Metadata.Bounds.Start)
+	assert.Equal(t, now.Add(20*time.Minute), m3CoordBlocks[1].Metadata.Bounds.End)
+
+	assert.Equal(t, models.Tags{"same": "tag"}, m3CoordBlocks[0].Metadata.Tags)
+	assert.Equal(t, models.Tags{"same": "tag"}, m3CoordBlocks[1].Metadata.Tags)
+}
+
+func TestMultipleNamespacesSuccess(t *testing.T) {
+	now := time.Now()
```

Comment:
This is the same test as the previous one bar the length check. May be better to refactor the test to a common function, then do the length check at the end?